File: 8kyu/(8 kyu) Remove First and Last Character/(8 kyu) Remove First and Last Character.cpp
#include <string>

using namespace std;

// 1 Simple solution
string sliceString (string str )
{
  string result = "";
    for ( int i = 1 ; i < str.length() - 1 ; i++ )
    {
      result += str[i] ;
    }
  return result ;
}

// 2 Optimized solution
string sliceString (string str )
{
  if ( str.length() <= 2 ) return "";
  return str.substr(1, str.length() - 2);
}

// 3 Coding golf
string sliceString (string str ) {
  return str.substr(1, str.length() - 2);
}

File: 8kyu/(8 kyu) Remove First and Last Character/(8 kyu) Remove First and Last Character.ts
// 1 Plain solution
export function removeChar(str: string): string {
    let result = '';

    for (let i = 1; i < str.length - 1; i++) {
        result += str[i];
    }

    return result;
}

// 2 Optimized solution
export function removeChar(str: string): string {
    return str
        .split('')
        .slice(1, -1)
        .join('');
}

// 3 Clever solution
export function removeChar(str: string): string {
    return str.slice(1, -1);
}

// 4 Coding golf
export const removeChar=(s:string):string=>s.slice(1,-1);
